src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
+                      [(and x-uflow? y-oflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; inf * 0 ~ nan, but if they rescue each other, then it should not be nan
+                      [(and x-oflow? y-uflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x underflows and y is normal, then if x != 0, then error
+                      [(and x-uflow? (not (= (abs subexpr-val) 0.0)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x overflows and y is normal, then if x != inf, then error
+                      [(and x-oflow? (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x is normal and y underflows, then if x != 0, then error
+                      [(and y-uflow? (not (= (abs subexpr-val) 0.0)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x is normal and y overflows, then if x != inf, then error
+                      [(and y-oflow? (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+                   
+                   [(list 'log.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (hash-ref exacts-hash arg))
+                    (define cond-num (abs (/ 1 subexpr-val)))
+                    (define arg-oflow? (overflow? arg))
+                    (define arg-uflow? (underflow? arg))
+                    (cond
+                      ; Condition number hallucination:
+                      ; Condition number is high when x = 1,
+                      ; but x is exactly 1, so there is no error
+                      [(and (= arg-val 1.0) (= subexpr-val 0.0)) #f]
+                      
+                      ; overflow rescue:
+                      [arg-oflow? (mark-erroneous! subexpr pt)]
+                      
+                      ; underflow rescue:
+                      [arg-uflow? (mark-erroneous! subexpr pt)]
+
+                      ; High Condition Number:
+                      ; CN(log, x) = |1 / log(x)|
+                      [(> cond-num 1e2) (mark-erroneous! subexpr pt)]
+                      [else #f])]
+                   
+                   [(list 'exp.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (hash-ref exacts-hash arg))
+                    
+                    (cond
+                      ; Condition Number Hallucination:
+                      ; When x is large enough that exp(x) overflows, condition
+                      ; number is also high.
+                      [(and (infinite? subexpr-val) (infinite? (exp arg-val)))
```

Comment:
Why do we need second condition here?

Reply:
Just to ensure that R(e^f(x)) also overflows, but yeah it is redundant. I am still inclined to keep it just in case.